Update user guide (#2130)
alan-turing-institute/data-safe-haven: docs/source/roles/researcher/using_the_sre.md

(roles_researcher_using_the_sre)=

# Using the Secure Research Environment

## {{newspaper}} Transferring files into or out of the SRE

Each time a request is made to bring data or software into ("ingress") or out of ("egress") the SRE, it needs to be reviewed in case it represents a security risk.
These reviews will be coordinated by the designated contact for your SRE.
They will have to discuss whether this is an acceptable risk to the data security with the project's principle investigator and data provider and the decision might be "no".

:::{hint}
You can make the process as easy as possible by providing as much information as possible about the code or data.
For instance, describing in detail what a dataset contains and how it will be use will help speed up decision making.
:::

## {{books}} Maintaining an archive of the project

:::{caution}
At the end of each project, the entire SRE is deleted.

> :::

Reviewer: This is true for us, but might not be for others.

> :::{caution}
> At the end of each project, the entire SRE is deleted.
> :::
> 
> Anything that has not been transferred to the **/output/** folder to be considered for egress will be deleted forever at this point.

Reviewer: ```suggestion
SREs are designed to be ephemeral and only deployed for as long as necessary.
It is likely that the infrastructure, and data, will be permanently deleted when work has concluded.

The `/output/` directory is designed for storing output to be kept after a project concludes.
You should move such data to the `/output/` directory and contact your designated contact about data egress.
```

:::{note}
You will not have permissions to install packages system-wide. You will need to use a **user package directory**.
:::

You can install **R** packages from inside **R** (or **RStudio**):

:::{code} R
> install.packages(NAME_OF_PACKAGE)
:::

You will see something like the following:

:::{code} R
Installing package into '/usr/local/lib/R/site-library'
(as 'lib' is unspecified)
Warning in install.packages("cluster") :
  'lib = "/usr/local/lib/R/site-library"' is not writable
Would you like to use a personal library instead? (yes/No/cancel)
:::

Type `yes`, which prompts you to confirm the name of the library:

:::{code} R
Would you like to create a personal library
'~/R/x86_64-pc-linux-gnu-library/4.1'
to install packages into? (yes/No/cancel)
:::

Type `yes` to install the packages.

(role_researcher_shared_storage)=

## {{open_file_folder}} Sharing files inside the SRE

There are several shared folder on each workspace that all collaborators within a research project team can see and access:

- [input data](#input-data): in the **/data/** folder
- [shared space](#shared-space): in the **/shared/** folder
- [output resources](#output-resources): in the **/output/** folder

> <!-- - [scratch space](#scratch-space-scratch): `/scratch/` -->

Reviewer: ```suggestion
```

I don't think this is likely to return.

> <!-- - [backup space](#backup-space-backup): `/backup/` -->

Author: ```suggestion
```

> <!-- - [backup space](#backup-space-backup): `/backup/` -->
> 
> ### Input data
> 

Author: ```suggestion
### Input data

```

Data that has been approved and brought into the secure research environment can be found in the **/data/** folder.

- The contents of **/data/** will be identical on all workspaces in your SRE.
- Everyone working on your project will be able to access it.
- Everyone has **read-only access** to the files stored here.

If you are using the Data Safe Haven as part of an organised event, you might find additional resources in the **/data/** folder, such as example slides or document templates.

:::{important}
You will not be able to change any of the files in **/data/**.
If you want to make derived datasets, for example cleaned and reformatted data, please add those to the **/shared/** or **/output/** folders.
:::

### Shared space

The **/shared/** folder should be used for any work that you want to share with your group.

- The contents of **/shared/** will be identical on all workspaces in your SRE.
- Everyone working on your project will be able to access it
- Everyone has **read-and-write access** to the files stored here.

> <!--
> #### Scratch space: `/scratch/`
> 
> The `/scratch/` folder should be used for any work-in-progress that isn't ready to share yet.
> Although everyone in your group will have **read-and-write access**, you can create your own folders inside `/scratch` and choose your own permissions for them.
> 
> :::{caution}
> You should not use `/scratch/` for long-term storage as it can be reset at any time without warning (_e.g._ when the VM is restarted).
> :::
> 
> The contents of `/scratch/` will be **different** on different VMs in your SRE.
> -->

Reviewer: ```suggestion
```

> <!--
> #### Backup space: `/backup/`
> 
> The `/backup/` folder should be used for any work-in-progress that you want to have backed up.
> In the event of any data loss due to accidental data deletion by a TRE user, your system administrator can restore the `/backup/` folder to the state it was in at an earlier point in time (up to 12 weeks in the past).
> This **cannot** be used to recover individual files - only the complete contents of the folder.
> Everyone in your group will have **read-and-write access** to all folders on `/backup`.
> 
> The contents of `/backup/` will be **identical** on all workspaces in your SRE.
> -->
> 
> ### Output resources

Author: ```suggestion
### Output resources
```

The **Gitea** server within the SRE can hold code, documentation and results from your team's analyses.
Use the **Gitea** server to work collaboratively on code with other project team members.

:::{important}
This **Gitea** server is entirely within the SRE - you do not need to worry about the security of the information you upload there as it is inaccessible from the public internet.
:::

You can access **Gitea** from an internet browser in the workspace using the desktop shortcut.
Use your **{ref}`short-form username <roles_researcher_username>`** and **password** to login.

::::{admonition} Logging in to Gitea
:class: dropdown note

- Click the **{guilabel}`Sign in`** button on the top-right of the page.

    :::{image} images/gitea_homepage.png
    :alt: Gitea homepage
    :align: center
    :width: 90%
    :::

- Enter your **{ref}`short-form username <roles_researcher_username>`** and **password**.

    :::{image} images/gitea_login.png
    :alt: Gitea login
    :align: center
    :width: 90%
    :::

- Then click the **{guilabel}`Sign in`** button

::::

::::{admonition} Create a new repository
:class: dropdown note

> - Log in to the **Gitea** dashboard

Reviewer: I would drop the Gitea documentation from here on and replace it with the Gitea docs.

Leave it with the specifics about credentials in the TRE and internal-only nature.

Author: The Gitea docs are actually pretty poor on most of these things. I'll have a go though.

Author: e.g. there's no equivalent of the Github [create a repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-new-repository) page.

Reviewer: In that case, lets keep this for now?
Maybe open an issue about finding an alternative.

Author: Captured in #2131
